### ISSUE TITLE
Slow Tab Switching (Images Related)

### ISSUE THREAD

**Issue by cameronhirbodi** [blackberry/bbUI.js#483]
Hi there!

I'm currently developing an app with eight tabs (four showing in the actionbar, the rest in the tab overflow). Though it runs choppy on my Dev A, a friends Dev B runs it very smoothly. So I'm pretty happy with the performance seeing it run well on a closer to production model. However, the one drawback I noticed on both devices is that switching between tabs can be very slow.

It's obvious that there is an inverse correlation between the amount of images and the speed of loading the tab. One tab has 12 ( 4 rows of 3) images using the square image grid. Each image is a 576 x 576px jpg. Loading this tab from another takes about 3-4 seconds. The other tab has 20 landscape images (10 rows of 2). Each of these images are larger at 799 x 449px for each jpg. This tab takes about 5-6 seconds. It hurts the flow of the app a bit when a user taps on the tab icon, sees it highlighted, but it takes several seconds to load.

So one of the ways I will try addressing this is shrinking the images while disabling landscape mode. This way they don't show larger and pixelated when the orientation is changed. Keeping it portrait only will allow me to keep them smaller.

I quickly mentioned this to you Tim, on Twitter, but each tab is a separate HTML page. I thought this worked better than keeping it all in one with separate id's, using "display:none" and controlling it through something like

function selectNews() {
    document.getElementById('news').style.display = 'block';
    document.getElementById('drivers').style.display = 'none';
    document.getElementById('teams').style.display = 'none';
}

In a nutshell, I'm asking what the best way to speed this up is. I think scaling down the images is one remedy but are there any others? Will there be more optimizing on the OS/Webworks that I should wait for or is there anything that can still be done on my part?

Thank you
